### ISSUE TITLE
Login through in Keycloak configured Identity Provider like Google are not working with SSL

### ISSUE THREAD

**Issue by stweiz** [nextauthjs/next-auth#3993]
### Provider type

Keycloak

### Environment

  System:
    OS: Linux 5.16 Arch Linux
    CPU: (12) x64 Intel(R) Core(TM) i7-9850H CPU @ 2.60GHz
    Memory: 7.62 GB / 15.26 GB
    Container: Yes
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.19.0 - /usr/bin/node
    Yarn: 1.22.17 - /usr/bin/yarn
    npm: 6.14.15 - /usr/bin/npm
  Browsers:
    Firefox: 97.0
  npmPackages:
    next: 10.0.7 => 10.0.7 
    next-auth: ^4.2.1 => 4.2.1 
    react: 17.0.1 => 17.0.1

### Reproduction URL

https://github.com/nextauthjs/next-auth-example

### Describe the issue

I have a Keycloak instance running, where I configured a Google identity provider. So I have two methods to login.

1. Login with password with an in Keycloak registered user.
2. Login through Google Auth.

As long as I use HTTP without SSL on local dev, the callback works. If I want to open my website with HTTPS, the first login method with password works. The second method through Google Auth is not working. I just get a 502.

Working:

Request URL: http://localhost:3000/api/auth/callback/keycloak?state=EzthHdHKz0JiffobPL_uk4NONSSyvmbhYNXDdZP7Wdk&session_state=a00f9baa-9709-44fd-ba91-39154d2e8c0f&code=0a516df3-0e53-4de0-95d5-2deb9a2261ff.a00f9baa-9709-44fd-ba91-39154d2e8c0f.d17026dd-82b0-4593-bc67-c4024a1b7aa1 (Status code 200)

Not working:

Request URL: https://control/api/auth/callback/keycloak?state=0ar7bR2D1EuqjA96Z6brPHZmC1n2HKE7wAzdsY4_U9o&session_state=a00f9baa-9709-44fd-ba91-39154d2e8c0f&code=1425d7a1-a899-47d0-9833-a0f7dfbe278d.a00f9baa-9709-44fd-ba91-39154d2e8c0f.d17026dd-82b0-4593-bc67-c4024a1b7aa1 (Status code 502)

Also there aren't any logs on the server or the browser console.

### How to reproduce

1. Setup Keycloak `docker run -p 8080:8080 jboss/keycloak`
2. Create a new client for next-auth in Keycloak.
3. Open `Identity Providers`.
4. Add `google` provider.
5. Enable SSL. I used nginx as reverse proxy with the following config to gain this.
`
openssl req -new -x509 -nodes -newkey rsa:4096 -keyout server.key -out server.crt -days 1095
chmod 444 server.crt
chmod 400 server.key
cp server.crt server.key /etc/nginx/ssl
`
**/etc/nginx/sites-enabled/custom_server.conf:**
`
server {
        listen 80;
        listen 443 ssl http2;
        server_name localhost;

        ssl_certificate ssl/server.crt;
        ssl_certificate_key ssl/server.key;

        location / {
                proxy_pass http://localhost:3000;
                proxy_set_header Host $http_host;
                proxy_set_header X-Forwarded-Host $http_host;
        }
}
`
5. Open next.js app with next-auth configured to this Keycloak instance.
6. Login.

Now you will be redirected to Keycloaks login page and can either login with username and password or with the Google Identity Provider. Both logins will work and now you will be redirected to the given callback url, which is the next.js app.

- The callback with the redirect URL http://localhost:3000/api/auth/callback/keycloak?state=EzthHdHKz0JiffobPL_uk4NONSSyvmbhYNXDdZP7Wdk&session_state=a00f9baa-9709-44fd-ba91-39154d2e8c0f&code=0a516df3-0e53-4de0-95d5-2deb9a2261ff.a00f9baa-9709-44fd-ba91-39154d2e8c0f.d17026dd-82b0-4593-bc67-c4024a1b7aa1 will work and you will see the next.js app.
- The callback with the redirect URL https://control/api/auth/callback/keycloak?state=0ar7bR2D1EuqjA96Z6brPHZmC1n2HKE7wAzdsY4_U9o&session_state=a00f9baa-9709-44fd-ba91-39154d2e8c0f&code=1425d7a1-a899-47d0-9833-a0f7dfbe278d.a00f9baa-9709-44fd-ba91-39154d2e8c0f.d17026dd-82b0-4593-bc67-c4024a1b7aa1 won't work and you won't see the next.js app. Just a 502 code and error page from nginx and no logs in the next.js app or the browser console.

### Expected behavior

The callback will work with HTTP and HTTPS.

**Comment by balazsorban44** [comment]
The connection between Keycloak and Google has nothing to do with `next-auth`, you set up Keycloak as a provider, not Google in your Next.js app, hence you don't see any logs in the Next.js app either.

This is not a NextAuth.js issue. Try using the Google provider directly in NextAuth.js, and see if that works. It might help you debug the configuration issues. :+1:

**Comment by Garciax** [comment]
@stweiz 
Faced the same problem. Have you been able to figure out what the problem is?

**Comment by stweiz** [comment]
@Garciax No, we decided to use another lib, because of this issue and the missing multi-tenancy ability.
